### PR TITLE
[#1597] Remove duplicate error messages in error summary

### PR DESCRIPTION
**Pull request by Zhou-Jiahao-1998**
Fixes https://github.com/reposense/RepoSense/issues/1597

Proposed commit message:
```
Duplicate error messages appear in the error summary shown in the
final report when the same repository has errors in multiple
configuration files.

Let's change the error summary data structure from List to Set to
remove duplicate error messages without affecting performance. The
order of error messages may change, but should not be a problem.
```

### CODE REVIEW

**Comment by gerhean** [comment]
Can you verify that your implementation works by creating a test case?

A set of mutable objects (map) sometimes have unexpected behavior (as the hash function is not properly defined).

**Comment by Zhou-Jiahao-1998** [comment]
@gerhean I hope I am addressing your question. According to [Java 11 Map](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/AbstractMap.html#equals(java.lang.Object)), two `Map` objects are equal if the mappings are identical. Since we only add into the set and do not modify the `Map` in any ways, we can treat each `Map` as immutable. I have added a test for a quick check.

**Comment by gerhean** [comment]
@Zhou-Jiahao-1998 By test I mean that you should try to recreate the situation where the duplicated error messages occurred, and ensure that the error messages indeed do not duplicate anymore.

**Comment by Zhou-Jiahao-1998** [comment]
@gerhean Alright I get what you mean now. Give me some time to figure out how to test this out without manually testing.

**Comment by gerhean** [comment]
Seems to be fine but I prefer that you verify that this solution works even if the problem repo is listed in the separate files of `author-config.csv` and `repo-config.csv`

**Review by dcshzj — CHANGES_REQUESTED** [review]
Per above

**Comment by Zhou-Jiahao-1998** [comment]
@dcshzj Took me some time to figure out how to do system test. I added an incorrect repo in both `author-config.csv` and `repo-config.csv` on the last line. This helps to check if the duplicated error is shown.

File: src/test/java/reposense/model/RepoLocationTest.java [hunk 1]
```diff
@@ -46,6 +47,55 @@ public void repoLocationParser_parseEmptyString_success() throws Exception {
         RepoLocation repoLocation = new RepoLocation("");
     }
 
+    @Test
+    public void duplicate_error_removed() {
```
Comment (dcshzj) [inline src/test/java/reposense/model/RepoLocationTest.java:51]: The function name should follow our coding conventions: `featureUnderTest_testScenario_expectedBehavior()`

**Review by dcshzj — CHANGES_REQUESTED** [review]

**Review by dcshzj — APPROVED** [review]
LGTM!

**Review by fzdy1914 — APPROVED** [review]

**Pull request merged**